quickwit-indexing/src/source/kinesis/api.rs

```diff
@@ -183,10 +187,46 @@ async fn list_streams(
     Ok(stream_names)
 }
 
+/// Merges two adjacent shards in a Kinesis data stream and combines them into a single shard.
+/// https://docs.aws.amazon.com/kinesis/latest/APIReference/API_MergeShards.html
+pub(crate) async fn merge_shards(
+    kinesis_client: &dyn Kinesis,
+    stream_name: &str,
+    shard_id: &str,
+    adjacent_shard_id: &str,
+) -> anyhow::Result<()> {
+    let request = MergeShardsInput {
+        stream_name: stream_name.to_string(),
+        shard_to_merge: shard_id.to_string(),
+        adjacent_shard_to_merge: adjacent_shard_id.to_string(),
+    };
+    // TODO: Implement retry.
+    kinesis_client.merge_shards(request).await?;
+    Ok(())
+}
+
+/// Splits a shard into two new shards in the Kinesis data stream.
+/// https://docs.aws.amazon.com/kinesis/latest/APIReference/API_SplitShard.html
+pub(crate) async fn split_shard(
```

Comment:
Is this something we trigger and we actually will do on quickwit? Or is an event we need to react to?

Reply:
This is an event we react to (see `NewShards` message). This is implemented here for testing purposes.

Reply:
I'm not sure I understand. Could it be #[cfg(test)] for instance?

Can we discuss it a little tomorrow maybe?

Reply:
Yeah, I added a few `#[cfg(test)]` in my `KinesisSource` branch after removing the `#![allow(dead_code)]`.

Reply:
Sorry about the confusion. I think we agree on this and will land on the same code, but I concede that the path I'm taking to get there is not the most straightforward.